pxr/usdImaging/usdImaging/collectionCache.cpp

```diff
@@ -38,10 +38,29 @@ _IsQueryTrivial(UsdCollectionAPI::MembershipQuery const& query)
         ruleMap.begin()->second == UsdTokens->expandPrims;
 }
 
+void
+UsdImaging_CollectionCache::_MarkCollectionContentDirty(UsdStageWeakPtr const& stage, 
+    UsdCollectionAPI::MembershipQuery const& query)
+{
+    if(query.HasExcludes())
+    {
+        // If there are any exlusion we have to consider all previously excluded paths dirty
+        // As there is no API to query excluded paths we mark all prims in the Stage dirty
+        _allPathsDirty = true;
```

Comment:
Even if the query has exclude paths, I'd still expect UsdComputeIncludedPathsFromCollection to come up with the right list.  That should return the set of paths that pass both the include & exclude rules, right?

Reply:
Indeed, but prims that were previously excluded from the collection would not be returned in the include set, and these also need to be marked dirty...

Reply:
Actually as the RemoveCollection call will catch all paths from the outgoing collection, the UpdateCollection should be able to safely ignore the excluded paths... Will test that further!